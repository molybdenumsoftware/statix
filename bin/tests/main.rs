mod util {
    #[macro_export]
    macro_rules! test_lint {
        ($tname:ident, $($tail:tt)*) => {
            test_lint!($tname);
            test_lint!($($tail)*);
        };
        ($tname:ident, $($tail:tt)*) => {
                test_lint!($tname);
                test_lint!($($tail)*);
        };
        ($tname:ident) => {
            paste::paste! {
                #[test]
                fn [<$tname _lint>](){
                    let file_path = concat!("tests/data/", stringify!($tname), ".nix");
                    test_cli(concat!(stringify!($tname), "_lint"), file_path, &["check"]);
                }

                #[test]
                fn [<$tname _fix>](){
                    let file_path = concat!("tests/data/", stringify!($tname), ".nix");
                    test_cli(concat!(stringify!($tname), "_fix"), file_path, &["fix", "--dry-run"]);
                }
            }

        };
    }
}

fn test_cli(test_name: &str, file_path: &str, args: &[&str]) {
    let output = std::process::Command::new("cargo")
        .arg("run")
        .arg("--")
        .args(args)
        .arg(file_path)
        .output()
        .expect("command runs successfully");

    let stdout = strip_ansi_escapes::strip(output.stdout).unwrap();
    let stdout = String::from_utf8(stdout).unwrap();

    insta::assert_snapshot!(test_name, &stdout);
}

test_lint! {
    collapsible_let_in,
    eta_reduction,
    useless_parens,
    empty_pattern,
    redundant_pattern_bind,
    unquoted_uri,
    empty_inherit,
    deprecated_to_path,
    useless_has_attr,
    repeated_keys,
    empty_list_concat
}
