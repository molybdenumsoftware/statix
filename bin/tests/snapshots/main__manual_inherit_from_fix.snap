---
source: bin/tests/main.rs
expression: "& stdout"
---
--- tests/data/manual_inherit_from.nix
+++ tests/data/manual_inherit_from.nix [fixed]
@@ -1,8 +1,8 @@
 let
   a = {b = 2; c = 3;};
 in
 {
-  b = a.b;
-  c = a.c;
+  inherit (a) b;
+  inherit (a) c;
 }
